### PR TITLE
DataGrid/TreeList: add the Column Chooser and Header Filter new features

### PR DESCRIPTION
**Pull request by sergepilipchuk**

### CODE REVIEW

File: api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/allowSelectAll.md [hunk 1]
```diff
@@ -5,7 +5,6 @@ default: false
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Allows users to simultaneously select all items.
```
Comment (vladaskorohodova) [inline api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/allowSelectAll.md]: It's true if this property is set to true. I would rephrase it to something like "Specifies whether to allow..."

File: api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/recursive.md [hunk 2]
```diff
@@ -5,7 +5,6 @@ default: false
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies whether selection is recursive.
```
Comment (vladaskorohodova) [inline api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/recursive.md:8]: Needs explanation what 'recursive' here means

File: api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/selectByClick.md [hunk 3]
```diff
@@ -5,7 +5,6 @@ default: false
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies whether or not an item becomes selected if a user clicks the item's label.
```
Comment (vladaskorohodova) [inline api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/selectByClick.md]: We usually do not say "whether or not", just "whether".

File: api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/enabled.md [hunk 4]
```diff
@@ -5,7 +5,6 @@ default: false
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies whether searching is enabled in the header filter.
```
Comment (vladaskorohodova) [inline api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/enabled.md]:
```suggestion
Specifies whether search is enabled in the header filter.
```

File: api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/timeout.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+id: HeaderFilterSearchConfig.timeout
+type: Number
+default: 500
+---
+---
+##### shortDescription
+Specifies a delay in milliseconds between a user finishes typing in the header filter's search panel, and the search start.
```
Comment (vladaskorohodova) [inline api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/timeout.md]: Doesn't feel like English. Rephrase

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+                // ...
+                search: {
+                    editorOptions: { 
+                        placeholder: 'Search column',
+                        mode: 'text',
+                        onOptionChanged: (e) => {
+                            if(e.name === 'onValueChanged') {
+                                // handle the option change here
+                            }
+                        }
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md]: you can use just onValueChanged instead of onOptionChanged

```suggestion
                        onValueChanged: (e) => {
                            // handle value change here
                        }
```

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+    import { Component } from '@angular/core'; 
+
+    @Component({ 
+        selector: 'app-root', 
+        templateUrl: './app.component.html', 
+        styleUrls: ['./app.component.css'] 
+    }) 
+
+    export class AppComponent { 
+        searchEditorOptions;
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md:55]: you need to set type for this field. Use type TextBoxConfiguration if possible, if not object
```suggestion
        searchEditorOptions: TextBoxConfiguration;
```

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+        constructor() {
+            this.searchEditorOptions = { 
+                placeholder: 'Search column', 
+                mode: 'text',
+                onOptionChanged: (e) => {
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md]: use just onValueChanged

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+    export default {
+        components: {
+            Dx{WidgetName}
+        },
+        data() {
+            return {
+                searchEditorOptions: { 
+                    placeholder: 'Search column', 
+                    mode: 'text',
+                    onOptionChanged: this.handleOptionChange
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md]: same here: use onValueChanged

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+                    :editor-options="searchEditorOptions"
+                    // ...
+                />
+            </DxColumnChooser>
+        </Dx{WidgetName}>
+    </template>
+
+    <script setup>
+    import 'devextreme/dist/css/dx.light.css';
+    import Dx{WidgetName}, {
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md:150]: Dx{WidgetName} should be inside brackets.
```suggestion
    import { Dx{WidgetName}
```

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+        DxColumnChooser, 
+        DxColumnChooserSearch,
+        // ... 
+    } from 'devextreme-vue/{widget-name}';
+
+    const searchEditorOptions = { 
+        placeholder: 'Search column', 
+        mode: 'text',
+        onOptionChanged: (e) => {
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md]: onValueChanged

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+    import {WidgetName}, {
+        ColumnChooser, 
+        ColumnChooserSearch, 
+        // ...
+    } from 'devextreme-react/data-grid';
+
+    const searchEditorOptions = { 
+        placeholder: 'Search column', 
+        mode: 'text',
+        onOptionChanged: (e) => {
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md]: onValueChanged

File: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/headerFilter/search/editorOptions.md [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+            // ...
+            fields: [{
+                // ...
+                headerFilter: {
+                    // ...
+                    search: {
+                        editorOptions: { 
+                            placeholder: 'Search value',
+                            mode: 'text',
+                            onOptionChanged: (e) => {
```
Comment (Tucchhaa) [inline api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/headerFilter/search/editorOptions.md]: onValueChanged

File: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/headerFilter/search/editorOptions.md [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+                // ...
+                fields: [{
+                    // ...
+                    headerFilter: {
+                        // ...
+                        search: {
+                            editorOptions: { 
+                                placeholder: 'Search value',
+                                mode: 'text', 
+                                onOptionChanged: (e) => {
```
Comment (Tucchhaa) [inline api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/headerFilter/search/editorOptions.md]: onValueChanged

File: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/headerFilter/search/editorOptions.md [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+        // ...
+        fields: [{
+            // ...
+            headerFilter: {
+                // ...
+                search: {
+                    editorOptions: { 
+                        placeholder: 'Search value',
+                        mode: 'text', 
+                        onOptionChanged: (e) => {
```
Comment (Tucchhaa) [inline api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/headerFilter/search/editorOptions.md]: onValueChanged

File: includes/column-header-filter-editor-options-code.md [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+            columns: [
+                {
+                    // ...
+                    headerFilter: {
+                        // ...
+                        search: {
+                            editorOptions: { 
+                                placeholder: 'Search city or state',
+                                mode: 'text',
+                                onOptionChanged: (e) => {
```
Comment (Tucchhaa) [inline includes/column-header-filter-editor-options-code.md]: onValueChanged

File: includes/header-filter-editor-options-code.md [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+    $(function(){
+        $("#{widgetName}").dx{WidgetName}({
+            // ...
+            headerFilter: {
+                // ...
+                search: {
+                    editorOptions: { 
+                        placeholder: 'Search value',
+                        mode: 'text',
+                        onOptionChanged: (e) => {
```
Comment (Tucchhaa) [inline includes/header-filter-editor-options-code.md]: onValueChanged

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+---
+##### jQuery
+
+    <!--JavaScript-->
+    $(function(){
+        $("#{widgetName}").dx{WidgetName}({
+            // ...
+            columnChooser: {
+                // ...
+                position: {
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md:22]: The snippet is correct, but the example is confusing. I would suggest using this:

`
position: {
  my: "right top",
  at: "right top",
  of: '.dx-datagrid'
}
`

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md [hunk 19]
```diff
@@ -0,0 +1,157 @@
...
+
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-{widget-name} ... >
+        <dxo-column-chooser ... >
+            <dxo-position
+                my="left"
+                at="right"
+                of=".dx-datagrid-column-chooser-button">
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md]: do you need to use ".dx-datagrid-column-chooser-button" class here? Shouldn't it be something like ".dx-{widgetname}-column-chooser-button"

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md [hunk 20]
```diff
@@ -119,15 +115,13 @@ See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxText
                 searchEditorOptions: { 
                     placeholder: 'Search column', 
                     mode: 'text',
-                    onOptionChanged: this.handleOptionChange
+                    onValueChanged: this.handleOptionChange
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md]: I would rename `this.handleOptionChange` to `this.handleValueChanged`

File: includes/header-filter-editor-options-code.md [hunk 21]
```diff
@@ -115,15 +111,13 @@
                 searchEditorOptions: { 
                     placeholder: 'Search value', 
                     mode: 'text',
-                    onOptionChanged: this.handleOptionChange   
+                    onValueChanged: this.handleOptionChange
```
Comment (Tucchhaa) [inline includes/header-filter-editor-options-code.md]: this.handleOptionChange -> this.handleValueChange

File: includes/column-header-filter-editor-options-code.md [hunk 22]
```diff
@@ -127,15 +123,13 @@
                 searchEditorOptions: { 
                     placeholder: 'Search city or state', 
                     mode: 'text',
-                    onOptionChanged: this.handleOptionChange
+                    onValueChanged: this.handleOptionChange
```
Comment (Tucchhaa) [inline includes/column-header-filter-editor-options-code.md]: this.handleOptionChange -> this.handleValueChange

**Review by Tucchhaa — CHANGES_REQUESTED** [review]

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md [hunk 23]
```diff
@@ -0,0 +1,157 @@
...
+    $(function(){
+        $("#{widgetName}").dx{WidgetName}({
+            // ...
+            columnChooser: {
+                // ...
+                position: {
+                    my: "left",
+                    at: "right",
+                    of: ".dx-datagrid-column-chooser-button"
+                }
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md]: The snippet is correct, but the example is confusing. I would suggest using this:

`position: { my: "right top", at: "right top", of: '.dx-datagrid' }`

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md [hunk 24]
```diff
@@ -0,0 +1,194 @@
+
+See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) for properties you can specify in this object.
```
Comment (albertov05) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/editorOptions.md]:
```suggestion
See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) topic for information about properties you can specify in this object.
```

File: api-reference/10 UI Components/GridBase/1 Configuration/headerFilter/search/editorOptions.md [hunk 25]
```diff
@@ -0,0 +1,11 @@
+
+See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) for properties you can specify in this object.
```
Comment (albertov05) [inline api-reference/10 UI Components/GridBase/1 Configuration/headerFilter/search/editorOptions.md]:
```suggestion
See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) topic for information about properties you can specify in this object.
```

File: api-reference/10 UI Components/dxPivotGrid/1 Configuration/headerFilter/search/editorOptions.md [hunk 26]
```diff
@@ -0,0 +1,8 @@
+
+See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) for properties you can specify in this object.
```
Comment (albertov05) [inline api-reference/10 UI Components/dxPivotGrid/1 Configuration/headerFilter/search/editorOptions.md]:
```suggestion
See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) topic for information about properties you can specify in this object.
```

File: api-reference/10 UI Components/dxPivotGridFieldChooser/1 Configuration/headerFilter/search/editorOptions.md [hunk 27]
```diff
@@ -0,0 +1,8 @@
+
+See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) for properties you can specify in this object.
```
Comment (albertov05) [inline api-reference/10 UI Components/dxPivotGridFieldChooser/1 Configuration/headerFilter/search/editorOptions.md]:
```suggestion
See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) topic for information about properties you can specify in this object.
```

File: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/headerFilter/search/editorOptions.md [hunk 28]
```diff
@@ -0,0 +1,212 @@
+
+See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) for properties you can specify in this object.
```
Comment (albertov05) [inline api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/headerFilter/search/editorOptions.md]:
```suggestion
See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) topic for information about properties you can specify in this object.
```

File: api-reference/40 Common Types/15 grids/ColumnChooserSearchConfig/enabled.md [hunk 29]
```diff
@@ -5,7 +5,6 @@ default: false
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies whether a search is enabled in the column chooser.
```
Comment (albertov05) [inline api-reference/40 Common Types/15 grids/ColumnChooserSearchConfig/enabled.md]:
```suggestion
Specifies whether search is enabled in the column chooser.
```

File: api-reference/_hidden/GridBaseColumn/headerFilter/search/editorOptions.md [hunk 30]
```diff
@@ -0,0 +1,11 @@
+
+See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) for properties you can specify in this object.
```
Comment (albertov05) [inline api-reference/_hidden/GridBaseColumn/headerFilter/search/editorOptions.md]:
```suggestion
See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) topic for information about properties you can specify in this object.
```

File: api-reference/_hidden/dxGanttHeaderFilter/search/editorOptions.md [hunk 31]
```diff
@@ -0,0 +1,8 @@
+
+See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) for properties you can specify in this object.
```
Comment (albertov05) [inline api-reference/_hidden/dxGanttHeaderFilter/search/editorOptions.md]:
```suggestion
See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) topic for information about properties you can specify in this object.
```

File: includes/column-header-filter-code.md [hunk 32]
```diff
@@ -0,0 +1,206 @@
...
+
+    <!-- tab: app.component.html -->
+    <dx-{widget-name} ... >
+        <dxi-column ... >
+            <dxo-header-filter ... >
+                <dxo-search
+                    [editorOptions]="searchEditorOptions"
+                    [enabled]="true"
+                    [timeout]="900"
+                    [searchExpr]="['City', 'State']"
```
Comment (Tucchhaa) [inline includes/column-header-filter-code.md]: can we move searchExpr array to variable as searchEditorOptions?

File: includes/column-header-filter-code.md [hunk 33]
```diff
@@ -0,0 +1,206 @@
...
+    <!-- tab: App.vue (Options API) -->
+    <template>
+        <Dx{WidgetName} ... >
+            <DxColumn ... >
+                <DxHeaderFilter ... >
+                    <DxSearch
+                        :editor-options="searchEditorOptions"
+                        :enabled="true"
+                        :timeout="900"
+                        :searchExpr="['City', 'State']"
```
Comment (Tucchhaa) [inline includes/column-header-filter-code.md]: +1

File: includes/column-header-filter-code.md [hunk 34]
```diff
@@ -0,0 +1,206 @@
...
+    <!-- tab: App.vue (Composition API) -->
+    <template>
+        <Dx{WidgetName} ... >
+            <DxColumn ... >
+                <DxHeaderFilter ... >
+                    <DxSearch
+                        :editor-options="searchEditorOptions"
+                        :enabled="true"
+                        :timeout="900"
+                        :searchExpr="['City', 'State']"
```
Comment (Tucchhaa) [inline includes/column-header-filter-code.md]: +1

File: includes/header-filter-editor-options-code.md [hunk 35]
```diff
@@ -0,0 +1,192 @@
...
+        HeaderFilter, 
+        Search, 
+        // ...
+    } from 'devextreme-react/data-grid';
+
+    const searchEditorOptions = { 
+        placeholder: 'Search value', 
+        mode: 'text',
+        onValueChanged: (e) => {
+            if(e.name === 'onValueChanged') {
```
Comment (Tucchhaa) [inline includes/header-filter-editor-options-code.md]: we do not need this if statement here

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md [hunk 36]
```diff
@@ -5,7 +5,154 @@ default: undefined
...
+    $(function(){
+        $("#{widgetName}").dx{WidgetName}({
+            // ...
+            columnChooser: {
+                // ...
+                position: {
+                    my: "left",
+                    at: "right",
+                    of: ".dx-{widgetname}-column-chooser-button"
+                }
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md]: Change this to
```suggestion
                position: {
                    my: "right top",
                    at: "right bottom",
                    of: ".dx-{widgetname}-column-chooser-button"
                }
```

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md [hunk 37]
```diff
@@ -5,7 +5,154 @@ default: undefined
...
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-{widget-name} ... >
+        <dxo-column-chooser ... >
+            <dxo-position
+                my="left"
+                at="right"
+                of=".dx-{widgetname}-column-chooser-button">
+            </dxo-position>
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md]: this too

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md [hunk 38]
```diff
@@ -5,7 +5,154 @@ default: undefined
...
+
+    <!-- tab: App.vue (Options API) -->
+    <template>
+        <Dx{WidgetName} ... >
+            <DxColumnChooser ... >
+                <DxPosition
+                    my="left"
+                    at="right"
+                    of=".dx-{widgetname}-column-chooser-button"
+                />
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md]: +1

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md [hunk 39]
```diff
@@ -5,7 +5,154 @@ default: undefined
...
+
+    <!-- tab: App.vue (Composition API) -->
+    <template>
+        <Dx{WidgetName} ... >
+            <DxColumnChooser ... >
+                <DxPosition
+                    my="left"
+                    at="right"
+                    of=".dx-{widgetname}-column-chooser-button"
+                />
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md]: +1

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md [hunk 40]
```diff
@@ -5,7 +5,154 @@ default: undefined
...
+
+    export default function App() { 
+        return ( 
+            <{WidgetName} ... > 
+                <ColumnChooser ... >
+                    <Position
+                        my="left"
+                        at="right"
+                        of=".dx-{widgetname}-column-chooser-button"
+                    />
```
Comment (Tucchhaa) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/position.md]: +1

**Review by Tucchhaa — CHANGES_REQUESTED** [review]

**Review by Tucchhaa — APPROVED** [review]

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/search.md [hunk 41]
```diff
@@ -0,0 +1,186 @@
+---
+id: GridBase.Options.columnChooser.search
+type: ColumnChooserSearchConfig
+inheritsType: ColumnChooserSearchConfig
+---
+---
+##### shortDescription
+Configures the column chooser's search.
```
Comment (Abadzhev) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/search/search.md]: "column chooser's search" is a vague phrase - is that a UI element, or API, or what are we even searching for? 

```suggestion
Configures the column chooser's search box.
```

or 

```suggestion
Configures the column chooser's search functionality.
```

File: api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/selection.md [hunk 42]
```diff
@@ -5,7 +5,7 @@ inheritsType: ColumnChooserSelectionConfig
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Configures the column chooser's selection.
```
Comment (Abadzhev) [inline api-reference/10 UI Components/GridBase/1 Configuration/columnChooser/selection.md]: Again, this is a vague phrase: "column chooser's selection". What do we select? Do we select the column chooser? Do we select columns? Is it a UI feature or runtime functionality? It's hard to say. 

```suggestion
Configures column selection functionality within the column chooser.
```

File: api-reference/10 UI Components/GridBase/1 Configuration/headerFilter/search/search.md [hunk 43]
```diff
@@ -0,0 +1,15 @@
+---
+id: GridBase.Options.headerFilter.search
+type: HeaderFilterSearchConfig
+inheritsType: HeaderFilterSearchConfig
+---
+---
+##### shortDescription
+Configures the header filter's search.
```
Comment (Abadzhev) [inline api-reference/10 UI Components/GridBase/1 Configuration/headerFilter/search/search.md]: Similar comment... Please update the rest of occurrences. 

```suggestion
Configures the header filter's search box.
```

or 

```suggestion
Configures the header filter's search UI.
```

File: api-reference/40 Common Types/15 grids/ColumnChooserSearchConfig/ColumnChooserSearchConfig.md [hunk 44]
```diff
@@ -6,7 +6,6 @@ type: Object
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Configures the column chooser's search.
```
Comment (Abadzhev) [inline api-reference/40 Common Types/15 grids/ColumnChooserSearchConfig/ColumnChooserSearchConfig.md]:
```suggestion
Configures the column chooser's search box.
```

or 

```suggestion
Configures the column chooser's search UI.
```

File: api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/ColumnChooserSelectionConfig.md [hunk 45]
```diff
@@ -6,7 +6,6 @@ type: Object
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Configures the column chooser's selection.
```
Comment (Abadzhev) [inline api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/ColumnChooserSelectionConfig.md]:
```suggestion
Configures column selection functionality within the column chooser.
```

File: api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/allowSelectAll.md [hunk 46]
```diff
@@ -5,7 +5,6 @@ default: false
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies whether to allow users to simultaneously select all items.
```
Comment (Abadzhev) [inline api-reference/40 Common Types/15 grids/ColumnChooserSelectionConfig/allowSelectAll.md]: It's not clear what you mean by "simultaneously" - the act of selection or the selection state:

- I can select all items with a single operation (let's say I press CTRL+A and select all ... simultaneously)
- I can select items one by one, and the option tells me if can ultimately select all of them. 

I guess it's the first option, because the second doesn't make much sense, but then I'd write something like this: 

 ```suggestion
Specifies whether a "Select All" option is available to users.
```

File: api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/enabled.md [hunk 47]
```diff
@@ -5,7 +5,6 @@ default: false
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies whether a search is enabled in the header filter.
```
Comment (Abadzhev) [inline api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/enabled.md]:
```suggestion
Specifies whether search UI is enabled in the header filter.
```

File: api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/timeout.md [hunk 48]
```diff
@@ -0,0 +1,10 @@
+---
+id: HeaderFilterSearchConfig.timeout
+type: Number
+default: 500
+---
+---
+##### shortDescription
+Specifies a delay in milliseconds between when a user finishes typing in the header filter's search panel, and when the search is executed.
```
Comment (Abadzhev) [inline api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/timeout.md]:
```suggestion
Specifies a timeout, in milliseconds, during which a user may continue to modify the search value without starting the search operation. 
```

File: api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/editorOptions.md [hunk 49]
```diff
@@ -0,0 +1,16 @@
...
+id: HeaderFilterSearchConfig.editorOptions
+type: any
+default: {}
+---
+---
+##### shortDescription
+Configures the search box.
+
+---
+See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) for properties you can specify in this object.
```
Comment (albertov05) [inline api-reference/40 Common Types/15 grids/HeaderFilterSearchConfig/editorOptions.md]:
```suggestion
See the [TextBox Configuration](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) topic for information about properties you can specify in this object.
```

**Review by vladaskorohodova — APPROVED** [review]
